modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -921,7 +921,7 @@ public void blockGateways() {
             GridCacheAffinityManager aff = cache.context().affinity();
 
             if (aff != null)
-                aff.cancelFutures();
+                aff.onKernalStop(true);
```

Comment:
Because, onKernalStop checks whether cache manager has started through variable **starting.** 
If manager hasn't started no NPE would be thrown. As a workaround i can create some method delegating cancelFutures and checking whether cache manager has started